src/IHaskell/Flags.hs

```diff
@@ -70,8 +71,8 @@ parseFlags flags =
   in case modeIndex of
     Nothing ->
       -- Treat no mode as 'console'.
-      if "--help" `elem` flags
-        then Left $ showText (Wrap 100) $ helpText [] HelpFormatAll ihaskellArgs
+      if "--version" `elem` flags || "-V" `elem` flags
+        then Left VERSION_ipython_kernel
```

Comment:
I investigated the `VERSION_` flags. They are defined by cabal for all packages. When you use `VERSION_ipython_kernel`, you're actually using the version for `ipython-kernel`, which might differ from the IHaskell version.

Looking at the old code, I think that the `if elem "--help" flags` bit was a hack to get help to show. The proper way of doing it would be to have the `parseFlags` function return a `Maybe Args` instead of `Either String Args`, and then working on the `Args` (if any) in the `ihaskell` function defined in `Main.hs`, as that is where all the flags are actually processed.